src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -318,4 +321,14 @@ public static function getRelativePath($basePath, $targetPath)
             || false !== ($colonPos = strpos($path, ':')) && ($colonPos < ($slashPos = strpos($path, '/')) || false === $slashPos)
             ? "./$path" : $path;
     }
+
+    private function hasLookAround($path)
+    {
+        return false !== strpos($path, '(?=') || false !== strpos($path, '(?<=') || false !== strpos($path, '(?!') || false !== strpos($path, '(?<!');
+    }
+
+    private function removeLookAround($path)
+    {
+        return preg_replace('/\\(\\?(=|<=|!|<!)((?:[^()]+|\\((?2)\\))*)[)]/', '', $path);
```

Comment:
what about escaped parenthesis in the pattern (yes, they're unlikely in routing, but not forbidden AFAIK) ?

Reply:
for reference, we have already have a regexp "parser" in RouteCompiler:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/RouteCompiler.php#L309 to deal with this kind of concern in another situation.

Reply:
Although I think it's unlikely, it is indeed possible. I think we can update the regex slightly to compensate for this: `...(?:[^()\\\\]+|\\\\.|...)...` (this will skip any escaped character while searching for the closing parenthesis)